packages/server/src/integrations/tests/utils/postgres.ts

```diff
@@ -1,10 +1,10 @@
 import { Datasource, SourceName } from "@budibase/types"
-import { GenericContainer, Wait, StartedTestContainer } from "testcontainers"
+import { GenericContainer, Wait } from "testcontainers"
 
-let container: StartedTestContainer | undefined
-
-export async function start(): Promise<StartedTestContainer> {
-  return await new GenericContainer("postgres:16.1-bullseye")
+export async function postgres(): Promise<Datasource> {
+  const container = await new GenericContainer("postgres:16.1-bullseye")
+    .withName("budibase-test-postgres")
+    .withReuse()
```

Comment:
This all looks good - one concern I have, just asking how it works, the `withReuse` - is it still tearing down the test containers correctly in dev environments after a test has ran? Just wondering if its leaving these containers running forever, while thats technically faster locally, it might cause some headaches if the containers are never reset.

Reply:
In my experience they do get left up. It's possible to tear them down, but I removed that code on request from @adrinr in a prior PR. I think there was a belief that `testcontainers` would stop them, but I have observed that's not the case. I'm happy to bring that code back, but doing so will lose the performance gain of having containers always running, which is substantial for tests that use them.

Reply:
I wonder if there is someway we could detect the teardown is needed outside of CI - my preference would be to teardown locally to avoid containers just running forever, which could lead to weirdness eventually as number of tables builds up etc. Could be worth investigating if we can just throw a teardown in if not in CI.

Reply:
What I've done instead is control it with an environment variable (`REUSE_CONTAINERS`). It's off by default and we turn it on in CI.